### PR TITLE
adds ReadMore component to Palette

### PR DESCRIPTION
**Pull request by ashleyjelks**
screen shot 2019-01-24 at 6 29 03 pm

FYI: We removed the track analytics decorator from the component when migrating to Palette, so this may need to be extended in Reaction to tack on the track decorator.

### CODE REVIEW

File: src/elements/ReadMore/ReadMore.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import React, { Component } from "react"
+import { renderToStaticMarkup } from "react-dom/server"
+import styled from "styled-components"
+import { DisplayProps } from "styled-system"
+import truncate from "trunc-html"
+
+export interface ReadMoreProps extends DisplayProps {
+  isExpanded?: boolean
+  maxChars?: number
+  onReadMoreClicked?: () => void
```
Comment (damassi) [inline src/elements/ReadMore/ReadMore.tsx]: This is typically where tracking is triggered from, so likely ok!

**Comment by damassi** [comment]
@ashleyjelks - Can you migrate the tests and docs from reaction as well?

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit e757b068e2a5a9e87326efa0f91a897a522ed5f4

https://deploy-preview-230--artsy-palette.netlify.com

**Comment by dblandin** [comment]
> @ashleyjelks - Can you migrate the tests and docs from reaction as well?

Weird... we definitely created a `docs/ReadMore.mdx` file porting over the storybook examples. Must not have committed it.

@ashleyjelks I know you're busy with some Local Discovery work. I'm happy to port over the specs and if you could push up the mdx file, that would be :100:.

**Comment by ashleyjelks** [comment]
> > @ashleyjelks - Can you migrate the tests and docs from reaction as well?
> 
> Weird... we definitely created a `docs/ReadMore.mdx` file porting over the storybook examples. Must not have committed it.
> 
> @ashleyjelks I know you're busy with some Local Discovery work. I'm happy to port over the specs and if you could push up the mdx file, that would be 💯.

@dblandin yep, I totally forgot to commit the `docs/ReadMore.mdx` file....just pushed it!

**Comment by dblandin** [comment]
@damassi Ready for another :mag_right:

**Comment by damassi** [comment]
Looks great 😄 Thanks y'all 👍

**Pull request merged**